### ISSUE TITLE
[Feature request] special "circles" for removing and blocking friends

### ISSUE THREAD

**Issue by zetok** [qTox/qTox#1958]
![blocked removed](https://cloud.githubusercontent.com/assets/3148759/8484621/86e85944-20f0-11e5-98a0-b28cce1ea7b1.png)
- [ ] circles are visible only when when there are members of them - i.e. if no one is blocked ~~/ removed~~, circles are not visible
- [ ] special circles are always on the bottom of friendlist
- [ ] user is not allowed to rename / remove / etc special circles
- [ ] messaging friends or any other form of contact shouldn't be possible
- [ ] friends in special circles have clickable `Unblock` (for `Blocked`) ~~and `Restore` (for `Removed`)~~ buttons in place of a status message
- [ ] friends in special circles have very limited options for r-click;
  -  in `Blocked`:
    - [ ] `Unblock friend`
      - removes from `Blocked` circle
      - [ ] when restoring, they should go back to their original group, if they were a part of one
    - [ ] ~~`Remove friend`  (moves to `Removed` circle)~~
  - in ~~`Removed`:~~
    - [ ] ~~`Restore friend`~~
      - [ ] ~~when restoring, they should go back to their original group, if they were a part of one~~
    - [ ] ~~`Permanently delete contact`~~
      - ~~deletes friend, should have a big red warning shown when clicked on it~~
- [ ] ~~Change `Remove friend` r-click option to move friend to `Removed` circle instead of actually removing it~~
  - [ ] ~~`Remove friend` shouldn't then remove history nor permanently delete friends~~
- [ ] ~~When permanently removing friend, warning should clearly state that recovering it will not be possible, and not be ambiguous as current one is, where it's not quite clear that when pressing `No` friend is still removed~~

**Comment by ProMcTagonist** [comment]
Proposition: Entries have a clickable Undo on them so the most obvious function can be done without right click.

**Comment by zetok** [comment]
@ProMcTagonist ooh, right, instead of a status message. Adding it to the list

**Comment by zetok** [comment]
What about:
`
“<zero-ghost> (…) having the actual list be in the privacy tab with a check box for showing the circle in the contact list”
`
?

Apparently some people don't want to have special circles shown, and would prefer to have it located in settings.

**Comment by romlok** [comment]
Opinions:

**Removed:**
I see little need to have a dedicated "removed" circle, if the removal confirmation dialog is made more explicit and clear.

**Blocked:**
After discussion in IRC, I feel that "blocked" would be a misleading term for what I understand to be the desired functionality - temporarily ignoring or silencing contacts. AFAICT the term "blocked" is used to mean something different - much more severe and permanent - in most other IM software, and the difference might lead to unnecessary misunderstanding by users.

Further, having a separate group for _temporarily_ ignored people seems excessive, when a simple indicator icon/toggle on the contact's entry could be sufficient.

**Comment by ProMcTagonist** [comment]
@zetok it would be neater having them only in settings, but the point of the removed list is to be able to quickly undo mistakes, yes?

**Comment by imconcerned** [comment]
perhaps changed "Blocked" to "Ignored"

**Comment by zetok** [comment]
@raumkraut 

> I see little need to have a dedicated "removed" circle, if the removal confirmation dialog is made more explicit and clear.

How do you "undo" removal then?

# 

@ProMcTagonist not really only for quick undoing of mistakes - removed contacts could go there, and stay there until user decides to remove them permanently.

E.g. one angrily removes someone from contacts, but a month later they want to actually contact that person (for whatever reason).

Point of it being available from friendlist, is to make it visible to users, so that they could know that they can undo removing / unignore someone. Locating it in settings wouldn't really give that, since most of target user-base doesn't touch the settings.

**Comment by tux3** [comment]
I'm with @raumkraut for the Removed list, I don't think it adds much. A dialog is enough confirmation for removing someone, and if you're prone to rage deleting people (is that a thing?) you can always add someone back later without sending a friend request as long as they didn't remove you too.

Also, the blocked and removed lists would work exactly the same internally, so having both seems a bit redundant. They both temporarily remove someone from the contacts (that's the only way to really block someone).

**Comment by zetok** [comment]
> I'm with @raumkraut for the Removed list, I don't think it adds much. A dialog is enough confirmation for removing someone, and if you're prone to rage deleting people (is that a thing?) you can always add someone back later without sending a friend request as long as they didn't remove you too.

How exactly do you add again someone who has been removed, since you didn't save their changed ID (why wouldn't they change nospam), nor it it possible to obtain it?

**Comment by tux3** [comment]
@zetok You're right, if they changed their nospam or if you lost their ID you can't add them back without asking.

I assumed you'd find find their ID the same way you found it the first time you added them, e.g. on toxme.se.

**Comment by zetok** [comment]
@tux3 the issue here is that it won't work with contacts that you've met somewhere e.g. travelling, exchanged IDs, and then you've removed them "because bad day, that's why", and a week later you decide that you didn't want to _actually remove_ them..

Looking through thousands of toxme entries for some ID that you don't remember, of a contact that changes their nick every other day?

Edit: also, why an assumption that they would use toxme?

**Comment by tux3** [comment]
@zetok 

> and then you've removed them "because bad day, that's why", and a week later you decide that you didn't want to actually remove them..

That's the thing that I wasn't accounting for. I'm generally not very impulsive, so I didn't think that was a thing. My instinct would be to just block them if I had a really bad day, not completely remove them.

**Comment by agilob** [comment]
@TheSpiritXIII is it doable with your implementation of circles?

**Comment by TheSpiritXIII** [comment]
You can use the base class that Circles uses for a possible implementation - CategoryWidget.

**Comment by aaannndddyyy** [comment]
-1 on having blocked **and** removed. 
It does the same thing. 
If you are too impulsive: either learn to control yourself, or make backups of the IDs.
Or just block the user. You can then undo that.
By default, removed friends could go into the blocked/removed circle.

**Comment by zetok** [comment]
Right. Since people don't seem to be interested in dedicated `Removed` circle, I'm crossing it out of the list.

> By default, removed friends could go into the blocked/removed circle.

Perhaps there should be some other place where removed contacts would go.. although I have no real idea where it could be, but using `Blocked` for that doesn't seem to be a good idea, since it would be confusing.

---

Is there any better way of implementing `Blocked`, considering that it should be:
1. Visible
2. Easy to access and use
3. Would be clear what it is for
4. other(?)

**Comment by aaannndddyyy** [comment]
somewhere else means either hidden and not quickly findable, not obvious, or again another circle called removed. I don't like the list being too cluttered. One system/special circle is enough, imho.

The only alternative I can think of, is have somewhere an undo button. 
But that would only undo the very last action.
EDIT: Re blkced/removed - if both words are used like this, people know immediately what this group is for and that also their removed contacts are in there. 
some might however dislike having to do an extra step for permanently deleting a peer.
If I delete, I want to delete it. Quickly, easily.

**Comment by anthonybilinski** [comment]
To address the implementation of actually blocking the friend (regardless of UI), it is possible to add a friend knowing only their public key, not their entire tox ID, so the friend changing nospam won't interfere with blocking/unblocking. Removing a friend at the core level, but having qTox remember their public key will stop them from being able to find your public key in the DHT, so they will have no idea if you're online and have no way to contact you, and then adding them back (without sending a friend request - in the same way we do when accepting a friend request by calling `tox_friend_add_norequest`) will allow both you and them to see each other once again.
